### PR TITLE
Prime SDK changes for the new backend services

### PR DESCRIPTION
**Pull request by kaushalrajbacancy**
## Description

-  The `DataUtils` module has been updated to fetch data from REST APIs provided by the new backend
- Added these endpoints into PrimeDataUtils: `getAccountBalances`, `getTransaction`, `getNftList`, `getAdvanceRoutesLiFi`, `getStepTransaction`, `getTokenLists`, `getTokenListTokens` and `fetchExchangeRates`
- Added new parameter named `chainId` in `getTokenLists` and `getTokenListTokens` endpoints

## Types of changes
What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Further comments (optional)

- Updated the `DataUtils` module to include a data API key parameter instead of project key and GraphQL endpoint parameters (If the data API key is not provided, it will automatically use the default API key, which has a strict rate limit)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,11 @@
 # Changelog
+## [1.5.3] - 2024-02-28
+### New
+- `PrimeDataUtils` has been added as a separate module, independent of the primeSdk, responsible for fetching data from REST APIs
```
Comment (ch4r10t33r) [inline CHANGELOG.md:4]: Pls call it DataUtils and replace the existing DataUtils with the prime backend.

File: examples/05-get-account-balances.ts [hunk 2]
```diff
@@ -1,11 +1,11 @@
-import { DataUtils, graphqlEndpoints } from '../src';
+import { PrimeDataUtils } from '../src';
 import * as dotenv from 'dotenv';
 
 dotenv.config();
 
 async function main() {
   // initializating Data service...
-  const dataService = new DataUtils('public-prime-testnet-key', graphqlEndpoints.QA)
+  const dataService = new PrimeDataUtils(process.env.DATA_API_KEY);
```
Comment (ch4r10t33r) [inline examples/05-get-account-balances.ts:8]: Please see above.

File: examples/06-transaction.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
-import { DataUtils, graphqlEndpoints } from '../src';
+import { PrimeDataUtils } from '../src';
 import * as dotenv from 'dotenv';
 
 dotenv.config();
 
 async function main(): Promise<void> {
   // initializating Data service...
-  const dataService = new DataUtils('public-prime-testnet-key', graphqlEndpoints.QA)
+  const dataService = new PrimeDataUtils(process.env.DATA_API_KEY);
```
Comment (ch4r10t33r) [inline examples/06-transaction.ts:8]: There should be a default key which have strict rate limitation.

**Review by ch4r10t33r — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (ch4r10t33r) [reply CHANGELOG.md:4]: You will be adding those in future versions of the SDK, no? Those using the endpoints you mentioned wont upgrade to your current version of prime sdk.

File: examples/06-transaction.ts [hunk 4]
```diff
@@ -1,11 +1,12 @@
-import { DataUtils, graphqlEndpoints } from '../src';
+import { DataUtils } from '../src';
 import * as dotenv from 'dotenv';
 
 dotenv.config();
+const dataApiKey = 'eyJvcmciOiI2NTIzZjY5MzUwOTBmNzAwMDFiYjJkZWIiLCJpZCI6IjI4ZWJiMGQ5YTMxYjQ3MmY4NmU4MWY2YTVhYzBhMzE1IiwiaCI6Im11cm11cjEyOCJ9';
```
Comment (ch4r10t33r) [inline examples/06-transaction.ts:5]: I would move this into DataUtils and use it as the default in case no key is passed.

**Review by ch4r10t33r — APPROVED** [review]
Pls make the minor change and then merge.

File: examples/06-transaction.ts (hunk 4 above)
Comment (kaushalrajbacancy) [reply examples/06-transaction.ts]: Done

**Review by 0xSulpiride — APPROVED** [review]

**Pull request merged**